lib/jxl/enc_ac_strategy.cc

```diff
@@ -1016,20 +1089,50 @@ void ProcessRectACSNew(PassesEncoderState* JXL_RESTRICT enc_state,
 
   // Priority is a tricky kludge to avoid collisions so that transforms
   // don't overlap.
-  uint8_t priority[64] = {0};
+  uint8_t priority[64] = {};
   for (auto tx : kTransformsForMerge) {
     AcStrategy acs = AcStrategy::FromRawStrategy(tx.type);
     for (size_t cy = 0; cy + acs.covered_blocks_y() - 1 < rect.ysize();
          cy += acs.covered_blocks_y()) {
       for (size_t cx = 0; cx + acs.covered_blocks_x() - 1 < rect.xsize();
            cx += acs.covered_blocks_x()) {
-        TryMergeToACSCandidate(tx.type, bx, by, cx, cy, config, cmap_factors,
-                               ac_strategy, tx.entropy_mul, tx.priority,
-                               &priority[0], entropy_estimate, block,
-                               scratch_space, quantized);
+        if (cy + 1 < rect.ysize() && cx + 1 < rect.xsize()) {
+          if (tx.type == AcStrategy::Type::DCT8X16) {
+            // We handle both DCT8X16 and DCT16X8 at the same time.
+            if ((cy | cx) % 2 == 0) {
+              Try16X8And8X16(bx, by, cx, cy, config, cmap_factors, ac_strategy,
+                             tx.entropy_mul, entropy_estimate, block,
+                             scratch_space, quantized);
+            }
+            continue;
+          } else if (tx.type == AcStrategy::Type::DCT16X8) {
+            // We handled both DCT8X16 and DCT16X8 at the same time,
+            // and that is above.
```

Comment:
Just remove it from `kTransformsForMerge`, maybe adding a comment there saying that 8x16 will also try 16x8?

Reply:
It is still used for the last column. If I remove compression gets significantly worse.

Reply:
I added more comments on this so that it may be now easier to follow.

Reply:
Oh right.